compiler/src/dmd/target.d

```diff
@@ -206,6 +208,7 @@ void addPredefinedGlobalIdentifiers(const ref Target tgt)
                     case 11: predef("FreeBSD_11"); break;
                     case 12: predef("FreeBSD_12"); break;
                     case 13: predef("FreeBSD_13"); break;
+                    case 14: predef("FreeBSD_14"); break;
```

Comment:
Why can't we use `FreeBSD_%d` here?

Reply:
That would be fine with me. I was just doing it the same way that it was already done. However, what would I use to format the string given that we can't use Phobos in the compiler code?

Reply:
snprintf can do this.

Reply:
Yeah. I figured that I'd ask in case there were some other way that something like that was typically done in the compiler, since I'm really not familiar with the dmd code base and what helper functions it has.

Reply:
Okay. It now uses `snprintf`. One other difference in behavior though is that it no longer defaults to FreeBSD 11 if the major OS version hasn't been set (which seems to typically be set by defining one of the TARGET_FREEBSD* versions when compiling dmd, and with these changes, build.d will now set it with the version of the OS being built on if it's not explicitly set). Instead, it just doesn't set the version-specific FreeBSD specifier, which will result in an error in druntime, which will make it easier to catch when we're missing a new version.